### PR TITLE
Add enumerations for PropTypes.oneOf usage

### LINKED ISSUE #234: Create enumeration for PropTypes.oneOf
# Issue Description
For [any package](https://github.com/cerner/terra-clinical/search?utf8=%E2%9C%93&q=oneOf%28&type=) that has a prop with a type of PropTypes.oneOf, an enumerations should be made available for that prop to make it easier for consumers to know the acceptable values. This also minimizes consumers from making typos when setting these props. 

An example of of a prop using enumeration is [alert type](https://github.com/cerner/terra-core/blob/master/packages/terra-alert/src/Alert.jsx#L16-L24) in the Alert component.

## Issue Type
- [ ] New Feature
- [ ] Enhancement
- [ ] Bug
- [X] Other

### PR DESCRIPTION
**Pull request by Blackop778**
### Summary
Adds an enumeration that is exported with components that has all the valid values for props that use `PropTypes.oneOf`. Also listed valid values in the prop's comment.

Closes #234

### CODE REVIEW

File: packages/terra-clinical-data-grid/src/DataGrid.jsx [hunk 1]
```diff
@@ -1103,5 +1103,7 @@ class DataGrid extends React.Component {
 
 DataGrid.propTypes = propTypes;
 DataGrid.defaultProps = defaultProps;
+DataGrid.Opts = {};
+DataGrid.Opts.ColumnSortIndicators = columnSortIndicators;
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/DataGrid.jsx]: Out of curiosity, why are we adding the Sort Indicator object to the DataGrid Opts? It doesn't look like we would need to use the sort indicator enum to render the DataGrid component?

File: packages/terra-clinical-item-collection/src/Item.jsx [hunk 2]
```diff
@@ -183,4 +203,13 @@ Item.defaultProps = defaultProps;
 Item.Display = ItemView.Display;
 Item.Comment = ItemView.Comment;
 
+const opts = {
+  Layouts: layouts,
+  EmphasisTypes: emphasisTypes,
+  Alignments: alignments,
+  Views: views,
```
Comment (emilyrohrbough) [inline packages/terra-clinical-item-collection/src/Item.jsx]: I am thinking these enum's names should be matching the name of the prop they are correlated with. For example: `Alignments ` to `accessoryAlignments`

File: packages/terra-clinical-item-view/src/ItemView.jsx [hunk 3]
```diff
@@ -190,6 +205,14 @@ const ItemView = ({
 
 ItemView.propTypes = propTypes;
 ItemView.defaultProps = defaultProps;
+
+const opts = {
+  Layouts: layouts,
+  EmphasisTypes: emphasisTypes,
+  Alignments: alignments,
```
Comment (emilyrohrbough) [inline packages/terra-clinical-item-view/src/ItemView.jsx]: I am thinking these enum's names should be matching the name of the prop they are correlated with. For example:` Alignments` to `AccessoryAlignments`

File: packages/terra-clinical-data-grid/src/DataGrid.jsx (hunk 1 above)
Comment (Blackop778) [reply packages/terra-clinical-data-grid/src/DataGrid.jsx]: The sort indicator is used in the `columnDataShape` in the `pinnedColumns` and `overflowColumns` props which are on the DataGrid component.

File: packages/terra-clinical-item-collection/src/Item.jsx (hunk 2 above)
Comment (Blackop778) [reply packages/terra-clinical-item-collection/src/Item.jsx]: This was done in https://github.com/cerner/terra-clinical/pull/510/commits/3bbac786827c62570528d93a7eae6ac15faf5314

File: packages/terra-clinical-item-view/src/ItemView.jsx (hunk 3 above)
Comment (Blackop778) [reply packages/terra-clinical-item-view/src/ItemView.jsx]: This was done in https://github.com/cerner/terra-clinical/pull/510/commits/3bbac786827c62570528d93a7eae6ac15faf5314

File: packages/terra-clinical-onset-picker/tests/jest/__snapshots__/OnsetPicker.test.jsx.snap [hunk 4]
```diff
@@ -52,6 +52,7 @@ exports[`should render a default onset picker with specified onset date 1`] = `
           aria-expanded="false"
           aria-haspopup="true"
           aria-labelledby="terra-select-screen-reader-label-1 terra-select-screen-reader-display-2 terra-select-screen-reader-placeholder-3"
+          aria-required="false"
```
Comment (Blackop778) [inline packages/terra-clinical-onset-picker/tests/jest/__snapshots__/OnsetPicker.test.jsx.snap]: Master is failing on this too, looks like a change in terra form select

File: packages/terra-clinical-data-grid/src/proptypes/columnDataShape.js [hunk 5]
```diff
@@ -27,13 +32,13 @@ const columnDataShape = PropTypes.shape({
   text: PropTypes.string,
   /**
    * A string indicating which sorting indicator should be rendered. If not provided, no sorting indicator will be rendered.
-   * If a `component` value is specified, `sortIndicator` will be ignored.
+   * If a `component` value is specified, `sortIndicator` will be ignored. One of `ascending`, `descending`.
    */
-  sortIndicator: PropTypes.oneOf(['ascending', 'descending']),
+  sortIndicator: PropTypes.oneOf([SortIndicators.ASCENDING, SortIndicators.DESCENDING]),
```
Comment (jmsv6d) [inline packages/terra-clinical-data-grid/src/proptypes/columnDataShape.js]: Could just do: 
```suggestion
  sortIndicator: PropTypes.oneOf(Object.values(SortIndicators)),
```

Comment (Blackop778) [reply packages/terra-clinical-data-grid/src/proptypes/columnDataShape.js]: Done in 4d40b4f2dccf8b009188b4a36fe230fd216d42c2

**Pull request merged**